devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy

```diff
@@ -0,0 +1,680 @@
+		childDevices.each {
+			def newLabel = getChildName(channelNumber(it.deviceNetworkId))
+			it.setLabel(newLabel)
+		}
+		state.oldLabel = device.label
+	}
+	initialize()
+}
+
+def initialize() {
+	runIn(3, "checkParam")
+}
+
+def installed() {
+	def numberOfButtons = 5
+	state.numberOfButtons = numberOfButtons
+	state.oldLabel = device.label
+	def existingChildren = getChildDevices()
+	for (i in 1..numberOfButtons) {
+		def buttonNetworkId = "${device.deviceNetworkId}:${i+2*numberOfButtons}"
+		def relayNetworkId = "${device.deviceNetworkId}:${i+numberOfButtons}"
+		def backlightNetworkId = "${device.deviceNetworkId}:${i}"
+		def childRelayExists = (existingChildren.find {child -> child.getDeviceNetworkId() == relayNetworkId} != NULL)
+		def childButtonExists = (existingChildren.find {child -> child.getDeviceNetworkId() == buttonNetworkId} != NULL)
+		def childBacklightExists = (existingChildren.find {child -> child.getDeviceNetworkId() == backlightNetworkId} != NULL)
+		if (!childBacklightExists ) {
+			def child = addChildDevice("Heltun Child Backlight", backlightNetworkId, device.hubId, [completedSetup: true, label: getChildName(i), isComponent: false])
+		}
+		if (!childRelayExists) {
+			addChildDevice("HELTUN", "Heltun Child Relay", relayNetworkId, device.hubId,[completedSetup: true, label: getChildName(i+numberOfButtons), isComponent: false])
+		}
+		if (!childButtonExists ) {
+			def child = addChildDevice("smartthings", "Child Button", buttonNetworkId, device.hubId, [completedSetup: true, label: getChildName(i+2*numberOfButtons), isComponent: true, componentName: "button$i", componentLabel: "Button ${i}"])
+		}
+	}
+	initialize()
+}
+
+private getChildName(channelNumber) {
+	def numberOfButtons = state.numberOfButtons
+	if (channelNumber in 1..numberOfButtons) {
+		return "${device.displayName} " + "${"Backlight"} " + "${channelNumber}"
+	}
+	else if (channelNumber in (numberOfButtons+1)..(2*numberOfButtons)){
+		return "${device.displayName} " + "${"Switch"} " + "${channelNumber-numberOfButtons}"
+	}
+	else if (channelNumber in (2*numberOfButtons+1)..(3*numberOfButtons)){
+		return "${device.displayName} " + "${"Button"} " + "${channelNumber-numberOfButtons*2}"
+	}
+}
+
+private channelNumber(String deviceNetworkId) {
+	deviceNetworkId.split(":")[-1] as Integer
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		return zwaveEvent(cmd)
+	}
+}
+
+private void setState(value, endpoint = null) {
+	def map = [
+		encap(zwave.basicV1.basicSet(value: value), endpoint),
+		encap(zwave.switchBinaryV1.switchBinaryGet(), endpoint),
+	]
+	sendHubCommand(map, 500)
+}
+
+private encap(cmd, endpoint) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+def childOn(childId) {
+	setState(0xFF, channelNumber(childId))
+}
+
+def childOff(childId) {
+	setState(0, channelNumber(childId))
+}
```

Comment:
It's silly, but it'd be nice to also be consistent "in style", so perhaps change `0`, to `0x00`?